### PR TITLE
Feat/legacy support

### PR DESCRIPTION
**Pull request by himanshuchawla009**

### CODE REVIEW

**Review by himanshuchawla009 — COMMENTED** [review]
- External function names should remain same
- Mark old internal functions with legacy prefix
- Invoke correct internal function based on network name and isMigrated flag.

File: src/helpers/metadataUtils.ts [hunk 1]
```diff
@@ -19,3 +25,57 @@ export async function decryptNodeData(eciesData: EciesHex, ciphertextHex: string
...
+
+export function generateMetadataParams(ecCurve: ec, serverTimeOffset: number, message: string, privateKey: BN): MetadataParams {
+  const key = ecCurve.keyFromPrivate(privateKey.toString("hex", 64));
+  const setData = {
+    data: message,
+    timestamp: new BN(~~(serverTimeOffset + Date.now() / 1000)).toString(16),
+  };
+  const sig = key.sign(keccak256(Buffer.from(stringify(setData), "utf8")).slice(2));
+  return {
+    pub_key_X: key.getPublic().getX().toString("hex"),
```
Comment (chaitanyapotti) [inline src/helpers/metadataUtils.ts:37]: pad to 64

File: src/torus.ts [hunk 2]
```diff
@@ -227,18 +203,563 @@ class Torus {
...
+        } catch {
+          throw new GetOrSetNonceError();
+        }
+        if (nonceResult.typeOfUser === "v1") {
+          modifiedPubKey = this.ec
+            .keyFromPublic({ x: X, y: Y })
+            .getPublic()
+            .add(this.ec.keyFromPrivate(nonce.toString(16)).getPublic());
+        } else if (nonceResult.typeOfUser === "v2") {
+          modifiedPubKey = this.ec
```
Comment (chaitanyapotti) [inline src/torus.ts:720]: should check if user upgraded or not

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**